x/observer/types/message_crosschain_flags_test.go

```diff
@@ -232,3 +234,67 @@ func TestMsgUpdateCrosschainFlags_GetRequiredPolicyType(t *testing.T) {
 		})
 	}
 }
+
+func TestMsgUpdateCrosschainFlags_GetSigners(t *testing.T) {
+	signer := sample.AccAddress()
+	tests := []struct {
+		name   string
+		msg    *types.MsgUpdateCrosschainFlags
+		panics bool
+	}{
+		{
+			name: "valid signer",
+			msg: types.NewMsgUpdateCrosschainFlags(
+				signer,
+				true,
+				true,
+			),
+			panics: false,
+		},
+		{
+			name: "invalid signer",
+			msg: types.NewMsgUpdateCrosschainFlags(
+				"invalid",
+				true,
+				true,
+			),
+			panics: true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if !tt.panics {
+				signers := tt.msg.GetSigners()
+				assert.Equal(t, []sdk.AccAddress{sdk.MustAccAddressFromBech32(signer)}, signers)
+			} else {
+				assert.Panics(t, func() {
+					tt.msg.GetSigners()
+				})
+			}
+		})
+	}
+}
+
+func TestMsgUpdateCrosschainFlags_Type(t *testing.T) {
+	msg := types.MsgUpdateCrosschainFlags{
+		Creator: sample.AccAddress(),
+	}
+	assert.Equal(t, types.TypeMsgUpdateCrosschainFlags, msg.Type())
+}
+
+func TestMsgUpdateCrosschainFlags_Route(t *testing.T) {
+	msg := types.MsgUpdateCrosschainFlags{
+		Creator: sample.AccAddress(),
+	}
+	assert.Equal(t, types.RouterKey, msg.Route())
+}
+
+func TestMsgUpdateCrosschainFlags_GetSignBytes(t *testing.T) {
+	msg := types.MsgUpdateCrosschainFlags{
+		Creator: sample.AccAddress(),
+	}
+	assert.NotPanics(t, func() {
+		msg.GetSignBytes()
+	})
+}
```

Comment:
Should we add a validate function for `BlockHeaderVerificationFlags` as well ?

Reply:
no I think it makes sense to cover it under `TestMsgUpdateCrosschainFlags_GetRequiredPolicyType`